--- ../pass.backup	2012-11-30 12:19:56.000000000 +0000
+++ password-store.sh	2013-04-29 16:14:01.699928000 +0100
@@ -9,6 +9,12 @@
 ID="$PREFIX/.gpg-id"
 GIT_DIR="${PASSWORD_STORE_GIT:-$PREFIX}/.git"
 GPG_OPTS="--quiet --yes --batch"
+# Suggested encFS check. Leave ENCFS_DIR blank for no encFS
+ENCFS_DIR="${ENCFS_STORE_DIR:-$HOME/.password-store.encfs}"
+# Attempt to mount encFS, or leave it to the user:
+ENCFS_AUTO=true
+# Mount encfs with --idle option, so unmount after this many minutes of inactivity, 0 if mounted indefinitely:
+ENCFS_IDLE_TIME=0
 
 export GIT_DIR
 export GIT_WORK_TREE="${PASSWORD_STORE_GIT:-$PREFIX}"
@@ -134,6 +140,26 @@
 	command="show"
 fi
 
+# Lev's suggested part
+
+if [[ ! -z $ENCFS_DIR ]]; then
+    if ! mount | grep $PREFIX > /dev/null; then
+	if $ENCFS_AUTO ; then
+	    if [ ! $ENCFS_IDLE_TIME -eq 0 ]; then
+		encfs --idle=$ENCFS_IDLE_TIME $ENCFS_DIR $PREFIX
+		echo "encFS mounted with idle-time: $ENCFS_IDLE_TIME mins."
+	    else
+		encfs $ENCFS_DIR $PREFIX
+	    fi
+	else
+	    echo "encFS not mounted! exiting..."
+	    exit 1
+	fi
+    fi
+fi
+
+# END Lev's part
+
 case "$command" in
 	init)
 		reencrypt=0
